### PR TITLE
[13.0][MIG] account_reconcile_blue_lines_restriction

### PR DESCRIPTION
**Pull request by grindtildeath**
Migrated from:
 - [ ] #311

### CODE REVIEW

**Review by Tonow-c2c — APPROVED** [review]
Add in the Roadmap to add tests

File: account_reconcile_blue_lines_restriction/models/account_reconcile_model.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                        OR
+                        aml.date > company.account_bank_reconciliation_start
+                    )
+                    -- This line restricts propositions to lines on
+                    -- accounts allowing reconciliation
+                    AND account.reconcile IS TRUE
+                    OR (
+                        -- black lines appearance conditions
+                        account.reconcile IS TRUE
+                        AND aml.reconciled IS FALSE
```
Comment (i-vyshnevska) [inline account_reconcile_blue_lines_restriction/models/account_reconcile_model.py:34]: this condition should already be in the query? I do not really understand how it should work

**Review by i-vyshnevska — APPROVED** [review]
LG, some questions

File: account_reconcile_blue_lines_restriction/models/account_reconcile_model.py (hunk 1 above)
Comment (grindtildeath) [reply account_reconcile_blue_lines_restriction/models/account_reconcile_model.py:34]: Yes this part is already there, however it's inside this big `AND` condition: https://github.com/odoo/odoo/blob/13.0/addons/account/models/account_reconcile_model.py#L575-L592
Therefore we must redefine here this whole condition to add `AND account.reconcile IS TRUE`.

I must still add tests though.

**Comment by grindtildeath** [comment]
After doing some testing, we found out that this module was created to avoid the bug we discovered yesterday with @yvaucher : https://github.com/odoo/odoo/pull/51223

So we probably will not need it anymore once the fix is merged.

**Comment by grindtildeath** [comment]
Module not needed anymore after Odoo fix got merged

**Pull request closed without merging**